message/message-util/util/src/java/org/sakaiproject/message/util/BaseMessage.java

```diff
@@ -561,10 +561,6 @@ public boolean isMessageViewable(Message message) {
 
 		String siteId = entityManager.newReference(message.getReference()).getContext();
 
-		if (unlockCheck(SECURE_READ_DRAFT, siteService.siteReference(siteId))) {
-			return true;
-		}
-
 		if (message.getHeader().getDraft()) return false;
```

Comment:
does something like make more sense?
```
if (message.getHeader().getDraft()) {
    // draft permission check
    if (!unlockCheck(SECURE_READ_DRAFT, siteService.siteReference(siteId))) {
        return false;
 } else {
    // normal permission check
    if (!unlockCheck(SECURE_READ, siteService.siteReference(siteId))) {
        return false;
    }
}
```

Reply:
No. the isMessageViewable is used by the widgets, dashboard, etc to pull the current announcements. a draft announcement can be viewed in the announcements tool with a DRAFT next to it if you have the permission. but it should not show up in the feeds.